test/test_diagnostics.py

```diff
@@ -0,0 +1,88 @@
+import numpy as onp
+import pytest
+from numpy.testing import assert_allclose
+from scipy.fftpack import next_fast_len
+
+from numpyro.diagnostics import (
+    _fft_next_fast_len,
+    autocorrelation,
+    autocovariance,
+    effective_sample_size,
+    gelman_rubin,
+    hpdi,
+    split_gelman_rubin
+)
+
+
+@pytest.mark.parametrize('statistics, input_shape, output_shape', [
+    (autocorrelation, (10,), (10,)),
```

Comment:
Maybe we can add more chain dims to autocorrelation, autocovariance and hpdi to ensure that they are aggregating across the chain dim?

Reply:
Except for `hpdi` which there is no `axis` arg yet, the other two can use `axis` to specify draw dimension. Is that the thing you have in mind? I will add `axis` to hpdi.

Reply:
+1. That will be great. The reason why I'm thinking about this, is that it will be really straightforward to support serial chaining (I think you have probably already done some experiments with parallel chaining), and it will be nice if we could use these diagnostics out of the box.